### PR TITLE
scylla-sstable: reject duplicate sstable names

### PR DESCRIPTION
**Pull request by tchaikov**
before this change, `load_sstables()` fills the output sstables vector
by indexing it with the sstable's path. but if there are duplicated
items in the given sstable_names, the returned vector would have uninitialized
shared_sstable instance(s) in it. if we feed such a sstables to the
operation funcs, they would segfault when derferencing the empty
lw_shared_ptr.

in this change, we error out if duplicated sstable names are specified
in the command line.

an alternative is to tolerate this usage by initializing the sstables
vector with a back_inserter, as we always return a dictionary with the
sstable's name as the key, but it might be desirable from user's
perspective to preserve the order, like OrderedDict in Python. so
let's preserve the ordering of the sstables in the command line.

this should address the problem of the segfault if we pass duplicated
sstable paths to this tool.

### CODE REVIEW

**Review by denesb — APPROVED** [review]

**Comment by tchaikov** [comment]
The failing dtest does expect the ordering to be preserved, lemme update it first.

**Comment by tchaikov** [comment]
@denesb as an alternative of this fix, how about erroring out if duplicated paths are specified in the command line?

**Comment by denesb** [comment]
> @denesb as an alternative of this fix, how about erroring out if duplicated paths are specified in the command line? It is less awkward than setting another slot in the vector if it is already set.

Makes sense.

**Comment by tchaikov** [comment]
v2:

- instead of tolerating the duplicated sstable paths in the command line, reject them.

@denesb hi Botond, could you take another look?

File: tools/scylla-sstable.cc [hunk 1]
```diff
@@ -2937,8 +2938,13 @@ Validate the specified sstables:
 
         std::vector<sstables::shared_sstable> sstables;
         if (app_config.count("sstables")) {
+            const auto sstable_names = app_config["sstables"].as<std::vector<sstring>>();
+            if (std::set(sstable_names.begin(), sstable_names.end()).size() != sstable_names.size()) {
+                fmt::print(std::cerr, "Duplicated sstables specified\n");
```
Comment (denesb) [inline tools/scylla-sstable.cc:2943]: error processing arguments: duplicate sstable arguments found

Comment (tchaikov) [reply tools/scylla-sstable.cc]: incorporated the proposed error message. thought "duplicate" was a verb but now i see it is also an adjective. thank you! TIL.

**Comment by tchaikov** [comment]
v3:

- incorporate the proposed error message.

**Review by denesb — APPROVED** [review]

**Pull request closed without merging**